### DISCUSSION TITLE
Documentation example raises TS error

### DISCUSSION THREAD

**Discussion by finnan444** [react-hook-form/react-hook-form#5183] [Q&A] [answered]
**Describe the bug**
In **TS support** section of the documentation there is an [example](https://react-hook-form.com/ts#ControlRef) with useWatch. However if you click CODESANDBOX, example differs from provided in the documentation. Adding type to `useWatch` generic raises error.

**To Reproduce**

1. Go to https://codesandbox.io/s/control-2mg07
2. Add `const firstName = useWatch<FormValues["firstName"]>({` instead of `const firstName = useWatch({`
3. See TS error: _Type 'string' does not satisfy the constraint 'Record<string, any>'.ts(2344)_

**Codesandbox link (Required)**
https://codesandbox.io/s/control-2mg07

**Screenshots**
<details><summary>Error</summary>

![Screenshot from 2021-05-13 12-12-09](https://user-images.githubusercontent.com/28092100/118108694-f8a0bb80-b3e8-11eb-9120-d9196e215e33.png)

</details>

**Comment by bluebill1049** [comment]
you don't need to provide the generic, the value type will be check against the input name.

**Comment by lamhungypl** [comment, marked as answer]
That docs is for version 6, in react-hook-form@7, it should be
```javascript
  const firstName = useWatch<FormValues, 'firstName'>({
    control,
    name: "firstName",
    defaultValue: "default"
  });
```

by default, watch value infer type from `control` and `defaultValue`, don't need type `useWatch`

**Reply by finnan444** [reply]
Thanks for your explanation.
I suppose Documentation example or CodeSandbox example should be updated

**Reply by bluebill1049** [reply]
can you show me which example? I can get it updated.

**Reply by finnan444** [reply]
https://react-hook-form.com/ts#ControlRef

<details> <summary>Screenshot</summary>

![Screenshot from 2021-05-14 15-58-27](https://user-images.githubusercontent.com/28092100/118274110-68887200-b4cd-11eb-9eb0-47812a2b293d.png)

</details>

And also CodeSandbox attached to example

**Reply by bluebill1049** [reply]
thanks fixed

**Reply by finnan444** [reply]
Thank you for your prompt response!
